src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs

```diff
@@ -124,10 +124,16 @@ protected override void ProcessRecord()
             {
                 if (ShouldProcess(this.VMScaleSetName, VerbsData.Update))
                 {
-
                     string resourceGroupName = this.ResourceGroupName;
                     string vmScaleSetName = this.VMScaleSetName;
-                    BuildPatchObject();
+                    if (this.VirtualMachineScaleSet == null)
```

Comment:
I don't understand this execution logic.  This prevenets users from piping scale sets into this cmdlet for PATCH.

The logic we discussed was that if any of the PATCH propertires were set, we would use PATCH, otherwise PUT

Reply:
piping scenario works as expected even it calls PUT.  (There is no difference between PUT and PATCH for CRP.)

The piping scenario is included in the test.

$vmss2 = $vmss | Update-AzureRmVmss -ResourceGroupName $rgname -Name $vmssName -SkuCapacity 4;

Reply:
That is not the piping scenario.  Why would I pass in resource group name and VMscaleset name when piping? The piping scenario would be:

```powershell
$vmss2 | Update-AzureRmVmss -SkuCapacity 4
```

Also, I thought part of thsi was that the PATCH was faster than the put.